core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponentSpec.scala

```diff
@@ -1141,6 +1141,33 @@ class EntityComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers wit
     }
   }
 
+  it should "return false from doesAttributeNameAlreadyExist if the attribute name does not exist" in withMinimalTestDatabase { _ =>
+    withWorkspaceContext(testData.workspace) { context =>
+      val exists = runAndWait(entityQuery.doesAttributeNameAlreadyExist(context, "Pair", AttributeName.withDefaultNS("case2"))).get
+      assert(!exists)
+    }
+  }
+
+  it should "change the attribute name when renameAttribute is called with valid arguments" in withDefaultTestDatabase {
+    withWorkspaceContext(testData.workspace) { context =>
+      val rowsUpdated = runAndWait(entityQuery.renameAttribute(context, "Pair",
+        AttributeName.withDefaultNS("case"), AttributeName.withDefaultNS("case2")))
+      assert(rowsUpdated == 2)
+      val exists = runAndWait(entityQuery.doesAttributeNameAlreadyExist(context, "Pair", AttributeName.withDefaultNS("case2"))).get
+      assert(exists)
+    }
+  }
+
+  it should "change the attribute name and namespace when renameAttribute is called with an attribute with a new namespace" in withDefaultTestDatabase {
+    withWorkspaceContext(testData.workspace) { context =>
+      val rowsUpdated = runAndWait(entityQuery.renameAttribute(context, "Pair",
+        AttributeName.withDefaultNS("case"), AttributeName.fromDelimitedName("new_namespace:new_name")))
+      assert(rowsUpdated == 2)
+      val exists = runAndWait(entityQuery.doesAttributeNameAlreadyExist(context, "Pair", AttributeName.fromDelimitedName("new_namespace:new_name"))).get
+      assert(exists)
+    }
+  }
```

Comment:
oooh this raises a good point: we currently have a list of allowed namespaces, and reject any others. I question whether or not we want to keep that allow-list but that's out of scope for this PR … which means you'll need to validate the new namespace before proceeding with any renaming. `withAttributeNamespaceCheck` should handle this for you: https://github.com/broadinstitute/rawls/blob/8de18c92666a364ad2866714833057068bded6d1/core/src/main/scala/org/broadinstitute/dsde/rawls/util/AttributeSupport.scala#L23

Reply:
Adding this-